javaslang/src-gen/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,437 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+/*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*\
+   G E N E R A T O R   C R A F T E D
+\*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*/
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import javaslang.collection.List;
+import javaslang.concurrent.Future;
+
+import javaslang.Function3;
+import javaslang.Function4;
+import javaslang.Function5;
+import javaslang.Function6;
+import javaslang.Function7;
+import javaslang.Function8;
+import javaslang.control.Option;
+import javaslang.control.Try;
+import javaslang.concurrent.Future;
+
+/**
+ * Applicative helpers
+ *
+ * @author Daniel Dietrich
+ * @since 2.1.0
+ */
+public final class Applicative {
```

Comment:
re-read: http://stackoverflow.com/a/23346604/516188

in applicative each step just transforms its input. in monadic style you can look at previous results too, and even also change the rest of computation. here we always perform the same steps, and each step only looks at its input. So I think we don't need the power of monads, applicative is enough. We can implement this feature with monads (that's liftM2 in haskell) but applicatives are enough (ilftA2 in haskell).